WordPress/gutenberg
# Parse (area-specific) context in template-part block.

## What?
Similar to #45534, this PR is part of the solution from https://github.com/WordPress/wordpress-develop/pull/3560, which addresses https://core.trac.wordpress.org/ticket/56930. We need to pass the `$context` parameter to `wp_filter_content_tags()` in the `template-part` block.

## Why?
We should always pass a context parameter to `wp_filter_content_tags()`, except when there is a current filter running. This is not the case when parsing the block template though, which is why https://github.com/WordPress/wordpress-develop/pull/3560 already takes care of it for the overall template. It also does so for the `template-part` block, however since that can only be edited within upstream Gutenberg, I have opened this PR.

Including the template part area identifier is useful context for that function, as it can make different decisions based on that. For example, images in the header of a site are extremely likely to appear above the fold, while for images in the footer the opposite applies.

## How?
The implementation here goes a bit beyond the core PR https://github.com/WordPress/wordpress-develop/pull/3560, using a cleaner implementation: We should ensure that the `$area` string is one of the allowed area identifiers before passing it through in the context.

## Testing Instructions
Test this as part of https://github.com/WordPress/wordpress-develop/pull/3560:

1. Add a `header` template part with an image in it.
2. Add a `footer` template part with an image in it.
3. See how the image in the header is not lazy-loaded, while the one in the footer is lazy-loaded.
